src/util.rs

```diff
@@ -1,36 +1,61 @@
+use crate::config;
 use crate::db::DB;
-use crate::dir::Dir;
-use crate::env::Env;
-use crate::types::Epoch;
+use crate::dir::{Dir, Epoch};
+
 use anyhow::{anyhow, bail, Context, Result};
+
+use std::cmp::{Ordering, PartialOrd};
 use std::io::{Read, Write};
+use std::path::Path;
 use std::process::{Command, Stdio};
 use std::time::SystemTime;
 
-pub fn get_db(env: &Env) -> Result<DB> {
-    let path = env
-        .data
-        .as_ref()
-        .ok_or_else(|| anyhow!("could not locate database file"))?;
-    DB::open(path)
+#[cfg(unix)]
+pub fn path_to_bytes<P: AsRef<Path>>(path: &P) -> Option<&[u8]> {
+    use std::os::unix::ffi::OsStrExt;
+
+    Some(path.as_ref().as_os_str().as_bytes())
+}
+
+#[cfg(not(unix))]
+pub fn path_to_bytes<P: AsRef<Path>>(path: &P) -> Option<&[u8]> {
```

Comment:
Why are these `Option`s? Just so you can use `to_str()?`?

Reply:
This operation can fail on Windows systems, returning an `Option` forces the caller to `unwrap` it explicitly, or handle the error.